app/controllers/openid_application_controller.rb

```diff
@@ -54,18 +54,33 @@ def authenticate_token
       return true
     end
 
+    return false if @session.uuid.nil?
+
     @current_user = OpenidUser.find(@session.uuid)
   end
 
   def populate_payload_for_launch_scope
-    launch = fetch_smart_launch_context
-    token.payload[:launch] = base64_json?(launch) ? JSON.parse(Base64.decode64(launch)) : { patient: launch }
+    @session = Session.find(token)
+    if @session.nil?
+      ttl = token.payload['exp'] - Time.current.utc.to_i
+      launch = fetch_smart_launch_context
+      @session = build_cc_session(ttl, launch)
+      @session.save
+    end
+    token.payload[:launch] =
+      base64_json?(@session.launch) ? JSON.parse(Base64.decode64(@session.launch)) : { patient: @session.launch }
```

Comment:
SSOI Tokens will make it to here and break. The launch is not stored in their session.

Reply:
added support for storing launch in session for ssoi tokens.